src/core/ChainRequest.js

```diff
@@ -0,0 +1,56 @@
+import fetch from 'node-fetch';
+
+const api_port = '4000';
+const url = 'http://localhost:';
```

Comment:
don't leave the colon in the URL like this; it makes it an invalid URL. it's the sort of thing that doesn't seem like a big deal but is slightly annoying when you go to change it or make it a function parameter. just do
```js
url + ':' + port // OR
`${url}:${port}`
```

in `request` below